src/main/kotlin/org/rust/ide/folding/RsFoldingBuilder.kt

```diff
@@ -63,6 +63,21 @@ class RsFoldingBuilder : CustomFoldingBuilder(), DumbAware {
         val rightMargin: Int
     ) : RsVisitor() {
 
+        override fun visitMacro(o: RsMacro) {
+            o.macroBody?.let(::fold)
+        }
+
+        override fun visitMacroExpansion(o: RsMacroExpansion) = when {
+            o.lparen != null -> foldBetween(o, o.lparen, o.rparen)
+            o.lbrack != null -> foldBetween(o, o.lbrack, o.rbrack)
+            else -> foldBetween(o, o.lbrace, o.rbrace)
+        }
+
+        override fun visitMacro2(o: RsMacro2) = when {
+            o.lparen != null -> foldBetween(o, o.lparen, o.rparen)
+            else -> foldBetween(o, o.lbrace, o.rbrace)
```

Comment:
Before:
<img width="150" alt="before" src="https://user-images.githubusercontent.com/6079006/56704469-e5ddc000-6715-11e9-8dd6-a43f988f8891.png">
After:
<img width="180" alt="after" src="https://user-images.githubusercontent.com/6079006/56704912-a1ebba80-6717-11e9-930e-d3384ec18c88.png">

Maybe it would be better to fold the macro body even if `o.lparen != null`?

```rust
override fun visitMacro2(o: RsMacro2) {
    foldBetween(o, o.lparen, o.rparen)
    foldBetween(o, o.lbrace, o.rbrace)
}
```

Reply:
Done, tests updated accordingly.